### PR TITLE
Fixed MathML conversion for dot class and increased code coverage

### LINKED ISSUE #155: `dot` accent is not rendering
Currently, `dot x` render as "dot x" instead of:
dot-issue

### PR DESCRIPTION
**Pull request by suleman-uzair**
This pull request aims to
1. Enhance the **MathML** conversion process for the **dot** class, ensuring the accurate rendering of **dot** symbols in **MathML** format.
2. It focuses on increasing the code coverage.

closes #155

### CODE REVIEW

**Comment by ronaldtse** [comment]
@suleman-uzair is this ready to go? Thanks!

**Comment by suleman-uzair** [comment]
> @suleman-uzair is this ready to go? Thanks!

Yes, now it's ready.

File: spec/plurimath/mathml/parser_spec.rb [hunk 1]
```diff
@@ -1729,4 +1729,73 @@
...
+                <mstyle mathvariant="italic">
+                  <mi>W</mi>
+                </mstyle>
+              </mrow>
+              <mrow>
+                <mi>t</mi>
+              </mrow>
+            </msub>
+            <mathml:mo rspace="-0.35em"/>
+            <mfenced open="(" close=")" separators="">
```
Comment (ronaldtse) [inline spec/plurimath/mathml/parser_spec.rb:1749]: `mfenced` is a deprecated element that we shouldn't use?

File: spec/plurimath/mathml/parser_spec.rb [hunk 2]
```diff
@@ -1729,4 +1729,73 @@
       expect(formula).to eq(expected_value)
     end
   end
+
+  context "contains metanorma mathml string Exp #1" do
+    let(:exp) {
+      <<~MATHML
+        <math xmlns="http://www.w3.org/1998/Math/MathML">
+          <mathml:mstyle scriptminsize="6pt">
```
Comment (ronaldtse) [inline spec/plurimath/mathml/parser_spec.rb:1737]: Why is the namespace called `mathml` here?

File: spec/plurimath/mathml/parser_spec.rb [hunk 3]
```diff
@@ -1729,4 +1729,73 @@
...
+              <mrow>
+                <mstyle mathvariant="italic">
+                  <mi>W</mi>
+                </mstyle>
+              </mrow>
+              <mrow>
+                <mi>t</mi>
+              </mrow>
+            </msub>
+            <mathml:mo rspace="-0.35em"/>
```
Comment (ronaldtse) [inline spec/plurimath/mathml/parser_spec.rb:1748]: Why is this namespaced?

File: spec/plurimath/mathml/parser_spec.rb [hunk 4]
```diff
@@ -1729,4 +1729,73 @@
...
+            <mfenced open="(" close=")" separators="">
+              <mathml:mspace width="-0.15em"/>
```
Comment (ronaldtse) [inline spec/plurimath/mathml/parser_spec.rb:1750]: Same here?

File: spec/plurimath/mathml/parser_spec.rb [hunk 5]
```diff
@@ -1729,4 +1729,73 @@
...
+                    <mstyle mathvariant="italic">
+                      <mi>T</mi>
+                    </mstyle>
+                  </mrow>
+                  <mrow>
+                    <mn>90</mn>
+                  </mrow>
+                </msub>
+              </mrow>
+              <mathml:mspace width="-0.1em"/>
```
Comment (ronaldtse) [inline spec/plurimath/mathml/parser_spec.rb:1763]: Same here?

**Review by ronaldtse — CHANGES_REQUESTED** [review]
Questions remain in the review comments

**Comment by suleman-uzair** [comment]
@ronaldtse Yes, you're correct. Currently, we don't use the `mfenced` element for **MathML** conversion. However, during parsing, we still accept this element in user input. So, users can include the `mfenced` element in their input, but the output generated by **Plurimath** won't contain this element.
An example of this can be found in the **metanorma** repos available at: `mn-samples-bipm/site/documents/brochure-its90/brochure-its90-en.pdf.fo.debug.xml`.

**Comment by ronaldtse** [comment]
I see, got it. Let’s merge this. Thanks!

**Review by ronaldtse — APPROVED** [review]

**Pull request merged**